:closed_lock_with_key: Refine security headers

neilbmclaughlin opened the pull request.

neilbmclaughlin commented:
I've added and configured helmet https://helmetjs.github.io/ for the same settings as in beta health (with some modifications to the content security whitelisting).

Redirection of http traffic to https is done at the infrastructure level so I have not implemented 

``` app.use(enforce.HTTPS({ ...}));```

enforce.HTTPS caused a large number of tests to fail and so, if it is felt to be required, this will need resolving.

I have, however, implemented [Strict-Transport-Security](https://helmetjs.github.io/docs/hsts/) as this makes the redirection to HTTPS more secure. See [here](https://scotthelme.co.uk/hsts-the-missing-link-in-tls/) for the details.

neilbmclaughlin commented on config/express.js:
```diff
@@ -21,7 +21,64 @@ module.exports = (app, config) => {
+      styleSrc: [
+        '\'self\'',
+        'fast.fonts.net',
+        'https://dhrlmnmyf2njb.cloudfront.net/',
```
Note: This URL is hard coded in the CSS too. Have raised an issue to make it configurable through an env var.

st3v3nhunt commented:
@neilbmclaughlin PR #170 Fixes a couple of errors with inline scripts/styles.

There is an issue with Firefox on my machine - the page isn't rendering full screen but it isn't on the beta domain either. I'll raise a separate issue for that.

neilbmclaughlin commented:
Added commit 4fdab81 :heavy_check_mark: Prevent inline script/style execution refusal
Removed commit 7f3db3e :no_bicycles: Add Strict-Transport-Security header

neilbmclaughlin commented:
Ready for approval and merge

st3v3nhunt commented:
As agreed with Adam (not in the GitHub org?!?) the `Strict-Transport-Security` header has been removed.

The rating is now a B https://securityheaders.io/?q=https%3A%2F%2Fconnecting-to-services-pr-167.herokuapp.com%2Ffinders%2Ffind-help&hide=on&followRedirects=on

Issue #172 has been raised to tackle inline script content policy

st3v3nhunt commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add http security headers](https://trello.com/c/6aJCdJP6/561-add-http-security-headers)
